autogpt_platform/backend/backend/server/integrations/router.py

```diff
@@ -416,3 +418,72 @@ def _get_provider_oauth_handler(
         client_secret=client_secret,
         redirect_uri=f"{frontend_base_url}/auth/integrations/oauth_callback",
     )
+
+
+@router.get("/ayrshare/sso_url")
+async def get_ayrshare_sso_url(
+    user_id: Annotated[str, Depends(get_user_id)],
+) -> dict[str, str]:
+    """
+    Generate an SSO URL for Ayrshare social media integration.
+
+    Returns:
+        dict: Contains the SSO URL for Ayrshare integration
+    """
+    # Generate JWT and get SSO URL
+    client = AyrshareClient()
+
+    # Get or create profile key
+    profile_key = creds_manager.store.get_ayrshare_profile_key(user_id)
+    if not profile_key:
+        logger.info(f"Creating new Ayrshare profile for user {user_id}")
+        # Create new profile if none exists
+        profile = client.create_profile(title=f"User {user_id}", messaging_active=True)
+        if isinstance(profile, PostError):
+            logger.error(
+                f"Error creating Ayrshare profile for user {user_id}: {profile}"
+            )
+            raise HTTPException(
+                status_code=500, detail="Failed to create Ayrshare profile"
```

Comment:
```suggestion
                status_code=502_BAD_GATEWAY, detail="Failed to create Ayrshare profile"
```
`502_BAD_GATEWAY` is exported by `starlette.status` and makes for slightly easier reading than just `502`